src/components/ModuleConstraints.tsx

```diff
@@ -181,10 +182,11 @@ const ModuleConstraints: React.FC<ModuleConstraintsProps> = ({ modules, onModule
         onModulesChange(mods);
     }
 
-    function cancelErrorAfterInterval() {
+    function showErrorThenCancelAfterInterval(fn: (a: boolean) => void) {
+        fn(true);
         if (activeErrorCancelTimeout !== undefined) clearTimeout(activeErrorCancelTimeout);
```

Comment:
I think there's a small bug here with the two error messages (that I didn't account for when writing this style haha). To repro:

* Type "cs2222" (invalid mod) into the modules box but don't submit yet 
* Paste "https://nusmods.com/sem-1/sh" (invalid URL) into the populate module link box but don't submit yet
* Then submit both one after the other
* We expect both error boxes to disappear, since we share the same timeout for both, only 1 box is cleared in the end and the other error stays around forever

Reply:
I had a hacky fix [=, from the user perspective should be good enough if we assume they donot spam the add module / share link button with the wrong module / link HAHAHA

Reply:
Haha okay let me take a look!

Reply:
Ok even with decent spamming it works haha :)

Reply:
cooll then, I have also added the remove all modules button for completeness

Reply:
Great! I saw it. Unfortunately the header for the table is removed when a mobile layout is used, so the remove all button also gets removed there. I'll fix it!